test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -924,3 +1328,50 @@ func getTunedStructuredData(profile *performancev2.PerformanceProfile) *ini.File
 	Expect(err).ToNot(HaveOccurred())
 	return cfg
 }
+
+func getCPUs(numaCoreSiblings map[int]map[int][]int, coreKey int, cpu *string, sep string) {
+	for key := range numaCoreSiblings {
+		for _, c := range numaCoreSiblings[key][coreKey] {
+			*cpu += sep + strconv.Itoa(c)
+			sep = ","
+			delete(numaCoreSiblings[key], c)
+		}
+	}
+}
+
+func getRanges(cpuString string) string {
+	offlineCPUs := ""
+	res := strings.Split(cpuString, ",")
+	var cpuInts = []int{}
+	for _, v := range res {
+		ret, _ := strconv.Atoi(v)
+		cpuInts = append(cpuInts, ret)
+	}
+	sort.Ints(cpuInts)
+	n1 := (cpuInts[:len(cpuInts)/2])
+	n2 := (cpuInts[len(cpuInts)/2:])
+	cpu1 := fmt.Sprintf("%d%s%d", n1[0], "-", n1[len(n1)-1])
+	cpu2 := fmt.Sprintf("%d%s%d", n2[0], "-", n2[len(n2)-1])
+	offlineCPUs = cpu1 + "," + cpu2
+	return offlineCPUs
+
+}
+
+func getCpusfromReservedIsolated(reservedCpus string, isolatedCpus string) string {
```

Comment:
This function needs a new name and a comment explaining what it does. It might also be better if it uses the cpu list parser.

Note the input might not be sorted.

Reply:
Have removed the function totally and incorporated it in the actual test case.